cluster_traffic_capture/cluster_traffic_capture/gen_haproxy_cfg.py

```diff
@@ -0,0 +1,182 @@
+from dataclasses import dataclass
+from typing import List
+
+
+SPOA_LISTEN_PORT = 12345
+SPOA_LISTEN_URL = "127.0.0.1"
+
+
+@dataclass
+class HostAddress:
+    host: str
+    port: int
+
+    @classmethod
+    def from_str(cls, address_str: str):
+        try:
+            host, port = address_str.split(":")
+            return cls(host, port)
+        except (IndexError, ValueError):
+            raise ValueError(f"Invalid host:port combination: {address_str}")
+
+
+def _gen_global_config() -> str:
+    """
+    Generates the HAProxy global/default configuration.
+    """
+
+    return """
+global
+    # Logging configuration
+    log 127.0.0.1:514 local0 # syslogd, facility local0
+    log stdout format raw local0 debug # stdout too
+
+    # Turn on Master/Worker mode.  Required to use SPOE Mirroring.  While hardly an expert, I've attempted to configure
+    # this so that the HAProxy container will exit if a process associated w/ the proxy mechanism runs into trouble.
+    # This will ensure problems are surfaced quickly/obviously, possibly at the expense of resilience at the container
+    # level.
+    #
+    # The premise of master/worker is that it sets up a main process which "is" HAProxy from the perspective of the
+    # rest of the system, and the actual work of HAProxy is done by subprocesses that can be killed/restarted at-will
+    # without changing the process id.  This provides better integration w/ systemd's process management system, and
+    # more logically bundles the various aspects of HAProxy under one process umbrella (the main process).
+    master-worker # not using no-exit-on-failure so container dies if any subprocess dies
+    mworker-max-reloads 3 # Limit subprocess reloads to 3 before SIGTERM is sent
+    hard-stop-after 15s # If a subprocess is still running 15s after reload signal, send SIGTERM
```

Comment:
Not quite.  This is handling the situation where the master process receives a signal (SIGUSR1) to reload the subprocesses and they haven't closed out in 15 seconds.  In that case, it will send them a SIGTERM to ensure they close out regardless of what they happened to be doing.  This makes sure that a hanging subprocess doesn't hang the entire collection of processes.  In this case, the subprocesses would be the mirroring SPOA and the actual HAProxy server.